core/src/main/java/org/commonjava/maven/ext/core/state/DependencyState.java

```diff
@@ -52,13 +51,13 @@
     @SuppressWarnings("DeprecatedIsStillUsed")
     @Deprecated
     @ConfigValue( docIndex = "dep-manip.html#exclusions-and-overrides", deprecated = true )
-    private static final String DEPENDENCY_EXCLUSION_PREFIX = "dependencyExclusion.";
+    public static final String DEPENDENCY_EXCLUSION_PREFIX = "dependencyExclusion.";
```

Comment:
As per above, why change from private to public?

Reply:
This was in response to your comment about a "hard-coded string" which I replied to. The strings were not meant to be hardcoded in the sense that I wanted to use constants for those.

I think it should be possible (and preferred) in the code to refer to the names of properties by constants instead of having to manually lookup or remember what is being used and hardcode that string into the messages. They are already there, but not all are made public.

Property constants which are currently public are mostly used in tests only which means they're not being used in messages even though they are being correctly used for property lookup.

Public but unused:

`BOMInjectingState`: all
`RelocationState.PLUGIN_RELOCATIONS`
`RESTState`: all except  `RESTState.REST_URL`  which is used for value lookup in `Cli` log message

Public and used in tests:

`CommonState.STRICT_ALIGNMENT_IGNORE_SUFFIX`
`CommonState.EXCLUDED_SCOPES`
`DistributionEnforcingState`: all
`RelocationState.DEPENDENCY_RELOCATIONS`
`VersioningState`: all except `VersioningState.VERSION_OVERRIDE_SYSPROP` seems unused

Reply:
Ok, I'm not particularly bothered about making them public given that they static consants, I more curious on the reason for the change

Reply:
The reason is to be able to access the properties (map keys) as constants.

I went through the code to find why some are public and some aren't. It appears that if a test was written, then it was made public.

I would like to submit a separate PR to make them all public and find any hardcoded uses after this one.

Reply:
Yes, pretty much - I've kept things as self-contained and private until they needed to be revealed.